File: testdata/fixtures/javascript/vulnerable/hardcoded_secret.ts
import jwt from 'jsonwebtoken';
import Stripe from 'stripe';

// VULNERABLE: Hardcoded JWT signing secret
const secret = "super_secret_jwt_key_do_not_share_2024!";

// VULNERABLE: Hardcoded API key (Stripe format)
const stripe = new Stripe('sk_live_4eC39HqLyjWDarjtT1zdp7dc', { apiVersion: '2023-10-16' });

// VULNERABLE: Hardcoded database connection string with credentials
const mongoUri = "mongodb://admin:Str0ngP@ssw0rd123@prod-db.example.com:27017/shopdb";

// VULNERABLE: Hardcoded password
const password = "R3allyStr0ng#Passw0rd!@2024";

// VULNERABLE: Hardcoded API key assigned to apikey variable
const api_key = "ak_prod_9f8e7d6c5b4a3210fedcba9876543210abcdef";

export function signToken(userId: string): string {
  // VULNERABLE: jwt.sign with hardcoded secret string
  return jwt.sign({ sub: userId, role: 'user' }, 'my-hardcoded-jwt-secret-key-very-unsafe');
}

export function getStripeClient(): Stripe {
  return stripe;
}

export function connectDB(): string {
  return mongoUri;
}

File: testdata/fixtures/go/vulnerable/hardcoded_secret.go
package vulnerable

import (
	"database/sql"
	"net/http"

	_ "github.com/lib/pq"
)

// VULN: Hardcoded credentials in source code.
// Should trigger GTSS-SEC-001 (Hardcoded Password) and GTSS-SEC-004 (Connection String).

var apiKey = "sk_live_4eC39HqLyjWDarjtT1zdp7dc"

var secret = "mySuperSecretKey2024!@#"

func ConnectDB() (*sql.DB, error) {
	connStr := "postgres://admin:p4ssw0rd_prod!@db.internal.example.com:5432/appdb?sslmode=disable"
	return sql.Open("postgres", connStr)
}

func HandleWebhook(w http.ResponseWriter, r *http.Request) {
	token := r.Header.Get("Authorization")
	if token != "Bearer "+apiKey {
		http.Error(w, "unauthorized", http.StatusUnauthorized)
		return
	}
	w.Write([]byte("OK"))
}
